File: LeetCode Solutions/January 2022 LeetCoding Challenge/Sequential Digits/Sequential Digits.py
class Solution:
    def sequentialDigits(self, low: int, high: int) -> List[int]:
        digits = "123456789"
        result = []
        nlow = len(str(low))
        nhigh = len(str(high))
        for i in range(nlow, nhigh + 1):
            for j in range(0, 10 - i):
                num = int(digits[j: j + i])
                if num >= low and num <= high:
                    result.append(num)
        return result

File: LeetCode Solutions/January 2022 LeetCoding Challenge/Sequential Digits.java
class Solution {
    public List<Integer> sequentialDigits(int low, int high) {
        String digits = "123456789";
        List<Integer> result = new ArrayList<>();
        int nlow = String.valueOf(low).length();
        int nhigh = String.valueOf(high).length();
        for (int i=nlow; i<=nhigh; i++){
            for (int j=0; j<10-i; j++){
                int num = Integer.parseInt(digits.substring(j, j+i));
                if (num >= low && num <= high){
                    result.add(num);
                }
            }
        }
        return result;
    }
}